### PR TITLE
Paths related UI fixes

### PR DESCRIPTION
**Pull request by liyiy**
## Changes

*Please describe.*  
*If this affects the frontend, include screenshots.*  

tooltips:
Screen Shot 2021-10-05 at 10 22 42 AM

before:
Screen Shot 2021-10-04 at 4 33 00 PM
afteR:
Screen Shot 2021-10-04 at 4 32 54 PM

before: (things are off centered, padding is wrong)
Screen Shot 2021-10-05 at 1 06 03 AM
after:
Screen Shot 2021-10-05 at 9 34 17 AM

## How did you test this code?

*Please describe.*

### CODE REVIEW

**Comment by clarkus** [comment]
Screen Shot 2021-10-05 at 8 12 01 AM

What are the numbers in the parenthesis for each path card item? In this example `(4)` is what I'm referencing.

**Comment by neilkakkar** [comment]
Total number of people reaching this path item

**Comment by clarkus** [comment]
OK that placement differs from the designs. That was supposed to be displayed flush right and then swap out on hover to show the inline ellipsis menu

Screen Shot 2021-10-05 at 8 24 49 AM

Edit to clarify concerns - my major concern is that the total number of people hitting the node looks more like a secondary descriptor of the path instead of proper metric. All other metrics are bold, black, flush right for easy comparisons and such. I know we're trying to balance functionality here, but this  latest styling feels off target to me. If we could restore the previous treatment and maybe always show that value and the inline menu, it could be a decent compromise?

**Comment by liyiy** [comment]
@clarkus Can you clarify what you mean by that last sentence? Like include the people count and the dropdown options? The original concern with swapping out the people count for the dropdown ellipsis was that the interaction didn't seem natural and also because the path card was already doing so much on hover (dropdown info, expanded text, dropdown options).

**Comment by clarkus** [comment]
Screen Shot 2021-10-05 at 10 47 43 AM

Yeah I just mean that we'd show everything in the summary row  regardless of hover. In this case the hovered card is shown on the left, the default case is shown right. For each item we show a step count, the path item name, the total count, and then the inline menu. Early in this project, I wasn't tracking the total count for the path item at all - I was just repeating the completed / continued count here. That's why I felt it was ok to hide it on hover because we'd be describing it in more detail below. In this case, we're showing distinct metrics, so showing all consistently makes more sense. This does increase the path item card's width, which means we might just have to show fewer steps until we can make some more fundamental changes to the visualizations.

**Comment by liyiy** [comment]
Also the people counts are in blue in the hover info because users can open the persons modal from there and we wanted to make it more obvious I think? @EDsCODE 

So then I just added the parenthesis to it because that's how we show it in funnels with persons count and then percentages

**Comment by EDsCODE** [comment]
RE: the color. How about we keep it black as shown in the latest image and we indicate clickability by showing the cursor and a grey tint for the entire row when hovering the main card, continued bar, or drop off bar? Being able to click an entire bar would be more convenient for the user

**Comment by clarkus** [comment]
Can we launch a persons modal based on the total count for the path item? If so, then that would be blue as well. In funnels, any persons count that is tied to a persons funnel gets that blue treatment. In this case, I think @liyiy is right that the count the summary row should also be blue to indicate that it is actionable. I also changed the percentages to align with funnels a bit more based on @liyiy's feedback above. 

It'd look like this if the total person count was actionable:
Screen Shot 2021-10-05 at 11 18 53 AM 
https://www.figma.com/file/gQBj9YnNgD8YW4nBwCVLZf/PostHog-App?node-id=3471%3A29287

**Comment by liyiy** [comment]
Font, padding, centering updates:

Screen Shot 2021-10-05 at 9 35 13 PM

Hiding path cards for smaller nodes until you hover: 
In this example I set it to a higher height because I couldn't find localhost examples of vertical congestion but in the code it's set to a smaller value

https://user-images.githubusercontent.com/25164963/136126856-19474547-5391-47c4-9113-6adc090dfd2d.mov

**Comment by clarkus** [comment]
Posted an idea to figma for differentiating start / end / fixed path items. https://www.figma.com/file/gQBj9YnNgD8YW4nBwCVLZf?node-id=4081:31449#114451257

**Comment by liyiy** [comment]
@clarkus I ended up highlighting the start/end nodes on hover after pairing with Tiina and seeing her understanding of it, rather than highlighting all start/end nodes in general 

Decided to leave out the path card item border highlight for now because the code implementation for it is kind of gross, so I want to see how this looks like first 

https://user-images.githubusercontent.com/25164963/136252915-f51b32c7-ca63-4d48-9062-db54cce0d913.mov

**Comment by clarkus** [comment]
> @clarkus I ended up highlighting the start/end nodes on hover after pairing with Tiina and seeing her understanding of it, rather than highlighting all start/end nodes in general

Oh I misunderstood. I was meaning for fixed starting / ending path query options. When you constrain the viz to only show paths starting at path item A, distinguish that with a color. For start / ends in an unbounded path, the path item counts would indicate position. I'm not sure we want to put too much meaning into those?

**Comment by liyiy** [comment]
> > @clarkus I ended up highlighting the start/end nodes on hover after pairing with Tiina and seeing her understanding of it, rather than highlighting all start/end nodes in general
> 
> Oh I misunderstood. I was meaning for fixed starting / ending path query options. When you constrain the viz to only show paths starting at path item A, distinguish that with a color. For start / ends in an unbounded path, the path item counts would indicate position. I'm not sure we want to put too much meaning into those?

ohhh yes I can do that too 😄, I just undid the hover highlighting rn because I actually need to account for multiple end nodes

**Comment by EDsCODE** [comment]
Added highlighting for all sources not just the topmost source

https://user-images.githubusercontent.com/13127476/136262280-9f7d996e-04c9-4768-bd23-d37624e4982d.mov

File: frontend/src/scenes/paths/NewPaths.js [hunk 1]
```diff
@@ -194,22 +207,28 @@ export function NewPaths({ dashboardItemId = null, filters = null, color = 'whit
 
         link.append('path')
             .attr('d', Sankey.sankeyLinkHorizontal())
-            .attr('id', (d) => `path${d.index}`)
+            .attr('id', (d) => `path-${d.index}`)
             .attr('stroke-width', (d) => {
                 return Math.max(1, d.width)
             })
             .on('mouseover', (data) => {
```
Comment (EDsCODE) [inline frontend/src/scenes/paths/NewPaths.js:214]: We should also make the cards appear when you hover a link. Right now you really have to line up your cursor with the thin bar for the card to appear.

The cards should hide again when you move the mouse away not only when you mouse over a different node

Comment (clarkus) [reply frontend/src/scenes/paths/NewPaths.js:214]: Why does it need to be pinned? I was just noticing the same issue and wondering why it didn't collapse when leaving the path item.

**Review by EDsCODE — APPROVED** [review]
cool changes. The hover interaction for hidden items still feels somewhat off but we can do another iteration to figure out exactly how hiding/showing cards should behave

**Pull request merged**